### ISSUE TITLE
Batch Norm not working

### ISSUE THREAD

**Issue by typicalTYLER** [jimkon/Deep-Reinforcement-Learning-in-Large-Discrete-Action-Spaces#35]
I can attach details later, but was just curious if you had intended to use batch norm or just hadn't messed with it (or if batch norm didn't make sense for this architecture, although I wouldn't know why). I changed the wolp_agent interface to allow for batch normalization as it appeared as if it should work, but got some errors that I couldn't fix in the time I had to work on it this morning.

Thanks!

**Comment by jimkon** [comment]
Batch Norm was a feature on DDPG's implementation. My work has nothing to do with this feature so I never made any changes to it. I don't think that it causes any problems for this architecture. So I didn't want to remove it completely, in case someone wants to use it.

One thing that I changed and probably causes some errors to the batch norm is that I added a function on the critic_net.py, but I didn't add it to the critic_net_bn.py.

The function is the following:
```python
 def evaluate_critic(self, state_1, action_1):
        return self.sess.run(self.critic_q_model, feed_dict={self.critic_state_in: state_1, self.critic_action_in: action_1})
```
This function calls the critic NN (not the target critic) to evaluate the k-NNs actions in order to select (argmax) the action with the greatest value expectations, as the algorithm of the paper proposes. 

Adding this function will probably fix the errors.

**Comment by typicalTYLER** [comment]
Thanks for the response! I'll report back if I have any luck running experiments with batch norm with these changes.

**Comment by typicalTYLER** [comment]
That seemed to get it running without error, I just had to also add `self.is_training: False` to the feed_dict as well. However, I did experience a very very large slowdown in the time per episode:

```
Episode  0 / 9999 started...Reward:8.0 Steps:450 t:107015 (238/step) Cur avg=8
Episode  1 / 9999 started...Reward:43.0 Steps:450 t:356586 (792/step) Cur avg=26
Episode  2 / 9999 started...Reward:43.0 Steps:450 t:669176 (1487/step) Cur avg=31
```

Compared to without batchnorm:

```
Episode  0 / 9999 started...Reward:290.0 Steps:450 t:6610 (15/step) Cur avg=290
Episode  1 / 9999 started...Reward:171.0 Steps:450 t:7021 (16/step) Cur avg=230
Episode  2 / 9999 started...Reward:161.0 Steps:450 t:6931 (15/step) Cur avg=207
Episode  3 / 9999 started...Reward:141.0 Steps:450 t:7067 (16/step) Cur avg=191
Episode  4 / 9999 started...Reward:169.0 Steps:450 t:8350 (19/step) Cur avg=186
Episode  5 / 9999 started...Reward:157.0 Steps:450 t:7439 (17/step) Cur avg=182
```

So I'll probably mess with other hyper-parameters to try to get it to converge instead.
Thanks for the quick response!
